test-data/unit/check-typeddict.test

```diff
@@ -2012,6 +2012,155 @@ v = {bad2: 2}  # E: Extra key "bad" for TypedDict "Value"
 [builtins fixtures/dict.pyi]
 [typing fixtures/typing-typeddict.pyi]
 
+[case testOperatorContainsNarrowsTypedDicts_unionWithList]
+from __future__ import annotations
+from typing import assert_type, TypedDict, Union
+from typing_extensions import final
+
+@final
+class D(TypedDict):
+    foo: int
+
+
+d: D | list[str]
+
+if 'foo' in d:
+    assert_type(d, Union[D, list[str]])
+else:
+    assert_type(d, list[str])
+
+[builtins fixtures/dict.pyi]
+[typing fixtures/typing-typeddict.pyi]
+
+[case testOperatorContainsNarrowsTypedDicts_total]
+from __future__ import annotations
+from typing import assert_type, Literal, TypedDict, TypeVar, Union
+from typing_extensions import final
+
+@final
+class D1(TypedDict):
+    foo: int
+
+
+@final
+class D2(TypedDict):
+    bar: int
+
+d: D1 | D2
+opt_d: D1 | None
+
+if 'foo' in opt_d:
+    assert_type(opt_d, D1)
```

Comment:
I don't see a lot of `assert_type` usage in tests. I think it is preferable to use `reveal_type`. It is more conventional for out code base.

Reply:
We didn't have it until recently, but maybe it's actually better? I used reveal_type in this test at first and it was more verbose and harder to read.

Reply:
(That is, I'm not saying "let's rewrite everything", but it's definitely clearer for new tests.)

Reply:
Hm, maybe you are right. At least, these tests are not sensible to output changes.